### PR TITLE
Misc bug fixes

### PR DESCRIPTION
**Pull request by Doralitze**
* fix: shift filter not filling out output channels

### CODE REVIEW

File: src/dmx/pixel.cpp [hunk 1]
```diff
@@ -10,12 +10,12 @@ namespace dmxfish::dmx {
     [[nodiscard]] std::string pixel::str() {
 		std::stringstream ss;
         if (this->iluminance >= 0.) {
```
Comment (LRahlff) [inline src/dmx/pixel.cpp:12]: do we need still need the condition?
getHue etc. should convert it automatically.
and the output has side effects, should it not output the values directly, so we can better investigate other problems? Maybe with a comment, which parts is invalid (or valid)?

File: src/filters/filter_conversion.hpp [hunk 2]
```diff
@@ -226,7 +226,10 @@ COMPILER_SUPRESS("-Weffc++")
         }
 
         virtual void update() override {
-            input->pixel_to_rgb(r, g, b);
+            input->pixel_to_rgb(this->r, this->g, this->b);
+	    //this->r=(uint8_t) (this->input->getRed()*256)/65536;
+            //this->g=(uint8_t) (this->input->getGreen()*256)/65536;
+            //this->b=(uint8_t) (this->input->getBlue()*256)/65536;
```
Comment (LRahlff) [inline src/filters/filter_conversion.hpp:232]: if it gets valid it should be 255 and 65535

File: src/filters/filter_shift.hpp [hunk 3]
```diff
@@ -127,7 +143,8 @@ namespace dmxfish::filters {
                     values.at(i) = values.at(i-1);
                 }
                 values.at(0) = *input;
-                last_update = last_update + *switch_time;
+                //last_update = last_update + *switch_time;
+		last_update = *time;
```
Comment (LRahlff) [inline src/filters/filter_shift.hpp:147]: the old version was for a predictable long duration. Now with every update it waits a bit longer than  switch time.

File: src/filters/filter_shift.hpp [hunk 4]
```diff
@@ -104,30 +116,44 @@ namespace dmxfish::filters {
...
+            if (last_update + *switch_time <= *time) {
+                for (size_t i = values.size() - 1; i > 0; i--) {
+                    values.at(i) = values.at(i - 1);
                 }
                 values.at(0) = *input;
-                last_update = last_update + *switch_time;
+                //last_update = last_update + *switch_time;
+                if (*switch_time > 0){
+                    last_update = last_update + std::floor((*time - last_update) / *switch_time) * *switch_time;
+                }
```
Comment (LRahlff) [inline src/filters/filter_shift.hpp:149]: adding the switch time as often as it overshooted the switch time

File: src/filters/filter_shift.hpp [hunk 5]
```diff
@@ -104,30 +116,44 @@ namespace dmxfish::filters {
...
+                // alternative instead of the above
+//                if (*time < last_update + 2 * *switch_time){
+//                    last_update = last_update + *switch_time;
+//                }
```
Comment (LRahlff) [inline src/filters/filter_shift.hpp:153]: just the basic idea: if switch_time was overshooted (at least twice), then set the actual time as reference

File: test/conversion_filter_test.cpp [hunk 6]
```diff
@@ -138,6 +138,15 @@ BOOST_AUTO_TEST_CASE(test_color_to_rgb) {
     BOOST_TEST(*map.eight_bit_channels["test:r"] == testr, "value r in filter color_to_rgb should be " + std::to_string(testr) + " but is " + std::to_string(*map.eight_bit_channels["test:r"]));
     BOOST_TEST(*map.eight_bit_channels["test:g"] == testg, "value g in filter color_to_rgb should be " + std::to_string(testg) + " but is " + std::to_string(*map.eight_bit_channels["test:g"]));
     BOOST_TEST(*map.eight_bit_channels["test:b"] == testb, "value b in filter color_to_rgb should be " + std::to_string(testb) + " but is " + std::to_string(*map.eight_bit_channels["test:b"]));
+
+    in_channel = dmxfish::dmx::pixel(0.0, 0.0, 1.0);
+    fil.update();
+    testr = 85; // TODO shouldn't this be 255?
```
Comment (LRahlff) [inline test/conversion_filter_test.cpp:144]: For my idea yes it should be 255, but this was the basic idea of the hsiluminance color room. Now the white has the same intensity of 255 (in sum of r, g, b) as a red with a value of 255.

Comment (Doralitze) [reply test/conversion_filter_test.cpp:144]: Let's deal with this after coal.

**Review by Doralitze — COMMENTED** [review]
LGTM

**Review by LRahlff — COMMENTED** [review]
LGTM

**Review by LRahlff — APPROVED** [review]

**Pull request merged**